OloEditor/src/Panels/SceneHierarchyPanel.cpp

```
@@ -383,6 +615,176 @@ namespace OloEngine
+        if (auto* mesh = std::get_if<EmitMesh>(&emitter.Shape))
+        {
+            int primIdx = mesh->PrimitiveType;
+            if (ImGui::Combo("Mesh Primitive", &primIdx, EmitMeshPrimitiveNames, EmitMeshPrimitiveCount))
+            {
+                BuildEmitMeshFromPrimitive(*mesh, primIdx);
+            }
+            ImGui::Text("Triangles: %u", static_cast<u32>(mesh->Triangles.size()));
+        }
+    }
+
+    static void DrawParticleRenderingSection(ParticleSystemComponent& component, ParticleSystem& sys)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'DrawParticleRenderingSection' of similar type ('int &') are easily swapped by mistake

(bugprone-easily-swappable-parameters)